project_template/_base/project_name/apps/common/utils/enumeration.py

```diff
@@ -0,0 +1,124 @@
+class Enumeration:
```

Comment:
I've tried enums in 3.5 but I wasn't satisfied with it. There were not as much abilities as required. If someone has experience of successful using this things in commercail projects let's move this question into discussion on a next meeting.

I know that there is another method for making choices by direct definition in models:

```python
class SomeModel(models.Model):

	STATUS_NEW = ...
	STATUS_PENDING = ...

	STATUSES = (
		STATUS_NEW, 'New',
		STATUS_PENDING, 'New',
	)
```

But again I think that custom `Enumeration` class gives more flexibility.

Reply:
There is an article, which covers usage of Enums for Django.

https://hackernoon.com/using-enum-as-model-field-choice-in-django-92d8b97aaa63 @SergeyKubrak you ask about it.

Quotes, which I found valuable (especially first one):
> You may ask, “Why using Enum instead of the examples given in Django official documentation?”
>
>  - To distinct from any other values within sets of values. For example, we could simple define integer 1–7 for days in a week, what about we have another set of values its representation includes the subset of 1–7?
> - Central Repository of predefined values, Enumeration can be used across your entire program, not just your in your models.

It will require to create our own base class for choices, but it can be based on Enums instead of dict.
Please, review this option.

Reply:
@alexryabtsev I play a bit with enums and have to consider, that they are not useful as choices. So that provided class enum is better candidature.

Enums require helper class that will act in a similar way to mentioned Enumeration and, more important, post author argument

> To distinct from any other values within sets of values. For example, we could simple define integer 1–7 for days in a week, what about we have another set of values its representation includes the subset of 1–7?

is not valid. As field values will be not enums, but enums values and therefore comparing have to be done in an ugly way BlaChoices.DE.value == 'German'. Fuuuuuu. Separate EnumField can be a better alternative, but out of scope for this conversation.